src/rpc.ts

```diff
@@ -45,16 +48,54 @@ router.post('/', async (req, res) => {
       return rpcError(res, 500, 'Unable to authenticate your verification link', id);
     } else if (method === 'snapshot.unsubscribe') {
       if (verifyUnsubscribe(params.email, params.signature)) {
-        await unsubscribe(params.email);
+        await unsubscribe(params.email, params.subscriptions);
         return rpcSuccess(res, 'OK', id);
       }
 
       return rpcError(res, 500, 'Unable to authenticate your verification link', id);
+    } else if (method === 'snapshot.update') {
+      if (!Array.isArray(params.subscriptions)) {
+        return rpcError(res, 400, 'Invalid params', id);
+      }
+
+      if (verifyUpdate(params.email, params.address, params.subscriptions, params.signature)) {
+        await update(params.email, params.address, params.subscriptions);
+        return rpcSuccess(res, 'OK', id);
+      }
+
+      return rpcError(res, 500, 'Unable to authenticate your request', id);
     }
   } catch (e) {
     console.log(e);
     return rpcError(res, 500, e, id);
   }
 });
 
+router.post('/subscriber', async (req, res) => {
+  const { address } = req.body;
+
+  try {
+    const subscribers = await getAddressSubscriptions(address);
+    if (!subscribers || !subscribers[0]) {
+      return rpcError(res, 404, 'Address not found', address);
+    }
+
+    return res.json(JSON.parse(subscribers[0].subscriptions as string));
+  } catch (e) {
+    console.log(e);
+    return rpcError(res, 500, e, address);
+  }
+});
+
+router.get('/subscriptionsList', (req, res) => {
```

Comment:
At FE will be used only keys that will be mapped to translations. Maybe we can simplify this endpoint(just returning an array of keys)?

Reply:
You already got the keys in the objects if you need. The full object is used by envelop-ui, which does not have i18n